shotgun_api3/shotgun.py

```diff
@@ -1752,8 +1785,13 @@ def _response_errors(self, sg_response):
         """
 
         if isinstance(sg_response, dict) and sg_response.get("exception"):
-            raise Fault(sg_response.get("message",
-                "Unknown Error"))
+            
+            if sg_response.get("error_code") == 102:
+                raise AuthenticationFault(sg_response.get("message", "Unknown Authentication Error"))
```

Comment:
That feels less fragile.  Only thing I'd say is to comment on that 102 to say where it comes from.  Otherwise it is just a magic number.